volumes.cpp

```diff
@@ -839,16 +838,17 @@ class RecVolume : public SurfaceVolume {
   Vector3d base_center;
   Transform transform;
   double length, radius1, radius2;
+  bool facet;
 
 public:
-  RecVolume( const Vector3d& center_p, const Vector3d& axis, const Vector3d& v1, const Vector3d& v2 ) :
+  RecVolume( const Vector3d& center_p, const Vector3d& axis, const Vector3d& v1, const Vector3d& v2, bool facet_p ) :
```

Comment:
A big problem into which I'm running is that "length" is also used to move the cylinder into position (I think it starts centered at the end of the first vector, rather than having one of its ends there).  If I pass in an axis with a length big enough to fill the world, it gets moved so that it only extends in one direction.  Any idea how I can work around this?  If possible, I'd rather not dig into and mess around with the iGeom functions...